src/cpr-mask.helpers.js

```diff
@@ -158,14 +160,14 @@ export function checkForAddingOrReplacing(input, oldMask, inputStop, maskPattern
 }
 
 export function oldToNewMask(characters, oldMask, start, pattern, filler) {
-	let replacePattern = pattern.slice(start).replace(/[A1W*]/g, "")
+	let replacePattern = pattern.slice(start).replace(/[A1W*]/g, "");
 	let charBank = characters;
 	let newMask = oldMask.slice(0, start) + oldMask.slice(start).split("").map((char, index) => {
 		if (char === replacePattern[0]) {
+			if (charBank[0] === replacePattern[0]) charBank = charBank.slice(1)
```

Comment:
Since we're adding the remaining charBank characters at the end, we want to slice off the first item in the array when an input value character matches a mask character.